src/components/text-editor/prosemirror-adapter/plugins/link-plugin.ts

```diff
@@ -0,0 +1,214 @@
+                    if (
+                        (event.metaKey || event.ctrlKey) &&
+                        event.button === 0
+                    ) {
+                        return processModClickEvent(view, event);
+                    }
+
+                    return processClickEvent(view, event);
+                },
+            },
+        },
+        view: () => ({
+            update: (view) => {
+                updateLink(view, updateLinkCallback);
+            },
+        }),
+    });
+};
+
+const findStart = (doc, pos) => {
+    let start = pos;
+    while (start > 0) {
+        const before = doc.nodeAt(start - 1);
+        if (
+            before &&
+            before.isText &&
+            before.marks.some((mark) => mark.type.name === EditorMenuTypes.Link)
+        ) {
+            start--;
+        } else {
+            break;
+        }
+    }
+
+    return start;
+};
+
+const findEnd = (doc, pos) => {
+    let end = pos;
+    while (end < doc.content.size) {
+        const after = doc.nodeAt(end);
+        if (
+            after &&
+            after.isText &&
+            after.marks.some((mark) => mark.type.name === EditorMenuTypes.Link)
+        ) {
+            end++;
+        } else {
+            break;
+        }
+    }
+
+    return end;
+};
+
+const getLinkDataAtPosition = (view: EditorView, event: MouseEvent) => {
+    const pos = view.posAtCoords({ left: event.clientX, top: event.clientY });
+    if (pos) {
+        const node = view.state.doc.nodeAt(pos.pos);
+        if (
+            node &&
+            node.marks.some((mark) => mark.type.name === EditorMenuTypes.Link)
+        ) {
+            const linkMark = node.marks.find(
+                (mark) => mark.type.name === EditorMenuTypes.Link,
+            );
+            const href = linkMark.attrs.href;
+            const from = findStart(view.state.doc, pos.pos);
+            const to = findEnd(view.state.doc, pos.pos);
+            const text = view.state.doc.textBetween(from, to, ' ');
+
+            return {
+                href: href,
+                text: text,
+                from: from,
+                to: to,
+            };
+        }
+    }
+
+    return null;
+};
+
+const processModClickEvent = (view: EditorView, event: MouseEvent): boolean => {
+    const { href } = getLinkDataAtPosition(view, event);
+    if (href) {
+        window.open(href, '_blank');
+
+        return true;
+    }
+
+    return false;
+};
+
+const openLinkMenu = (view: EditorView, href: string, text: string) => {
+    const event = new CustomEvent<EditorLinkMenuEventDetail>(
+        'open-editor-link-menu',
+        {
+            detail: { href: href, text: text },
+            bubbles: true,
+            composed: true,
+        },
+    );
+    view.dom.dispatchEvent(event);
+};
+
+let lastClickTime = 0;
+const processClickEvent = (view: EditorView, event: MouseEvent): boolean => {
+    const DOUBLE_CLICK_DELAY = 500;
```

Comment:
We can place this constant outside the function

Reply:
⚡